### PR TITLE
Drop git dependency on enso-bot repository

### PR DESCRIPTION
**Pull request by Frizi**
Removes a git-dependency to one of our other repositories, which we will likely soon make private or remove. Instead, the required chat client API typedefs are moved into the dashboard codebase directly. This gets rid of a few problematic transitive dependencies that we don't actually need (e.g. `better-sqlite3`).

### CODE REVIEW

File: app/gui/src/dashboard/services/Chat.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+// =======================
+// === AWS Cognito API ===
+// =======================
+
+/** A user of the application. */
+// export interface User {
+//   id: string
+//   name: string
+//   email: EmailAddress
+// }
```
Comment (farmaazon) [inline app/gui/src/dashboard/services/Chat.ts]: Commented structure?

File: app/gui/src/dashboard/services/Chat.ts
Comment (farmaazon) [inline app/gui/src/dashboard/services/Chat.ts:1]: If these types were copied, can we add a reference from where were they copied?

**Review by farmaazon — APPROVED** [review]
I really like the number of lines removed from pnpm-lock.

**Review by somebody1234 — APPROVED** [review]

**Review by vitvakatu — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

**Comment by 4e6** [comment]
ydoc runs `pnpm -r compile` and fails with
```
 INFO ide_ci::program::command: sbt ℹ️ [info] app/ydoc-shared compile$ tsc
 INFO ide_ci::program::command: sbt ℹ️ [info] app/ydoc-shared compile: sh: 1: tsc: not found
 INFO ide_ci::program::command: sbt ℹ️ [info] /runner/_work/enso/enso/app/ydoc-shared:
 INFO ide_ci::program::command: sbt ℹ️ [info]  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  ydoc-shared@0.1.0 compile: `tsc`
 INFO ide_ci::program::command: sbt ℹ️ [info] spawn ENOENT
 ```
 At this point I'm not sure if it's an issue with the node upgrade or with the Demeter runner.

**Review by Akirathan — APPROVED** [review]

File: distribution/engine/THIRD-PARTY/NOTICE [hunk 2]
```diff
@@ -491,11 +306,6 @@ The license file can be found at `licenses/Universal_Permissive_License__Version
 Copyright notices related to this dependency can be found in the directory `org.graalvm.shadowed.xz-24.0.0`.
 
 
-'chromeinspector-tool', licensed under the GNU General Public License, version 2, with the Classpath Exception, is distributed with the engine.
-The license file can be found at `licenses/GNU_General_Public_License__version_2__with_the_Classpath_Exception`.
-Copyright notices related to this dependency can be found in the directory `org.graalvm.tools.chromeinspector-tool-24.0.0`.
```
Comment (radeusgd) [inline distribution/engine/THIRD-PARTY/NOTICE:496]: Aren't we distributing it, just in a different way?

**Review by radeusgd — APPROVED** [review]

File: distribution/engine/THIRD-PARTY/NOTICE (hunk 2 above)
Comment (Akirathan) [reply distribution/engine/THIRD-PARTY/NOTICE:496]: Yes, as `component/chromeinspector-tool-24.0.0.jar`. Seems like license review tool is unable to determine that. Let's not deal with it in this PR, we need to merge it ASAP.

Comment (radeusgd) [reply distribution/engine/THIRD-PARTY/NOTICE:496]: We already have a way of adding such 'implicit' dependencies, see `runtime-and-langs`.

I'm worried that if we don't address it now, we will forget about it. Let's at least create a ticket and try to do a follow up sooner than later.

**Pull request merged**